compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -200,14 +200,17 @@ impl<'tcx> AbstractConst<'tcx> {
         Ok(inner.map(|inner| AbstractConst { inner, substs: uv.substs }))
     }
 
-    pub fn from_const(
+    pub fn from_constant(
```

Comment:
`ConstantKind::Value` is for non type system stuff isnt it? all of the stuff in this file is to do with type level constants so it seems kinda sus to be taking a `ConstantKind` here instead of a `Const`. I'm not all that familiar with `mir::Constant` so I may be off here :thinking:

Reply:
Yes `ConstantKind::Value` isn't meant to be seen by the type system, but I don't think this is a problem here since we basically discard those here and only care about unevaluated constants. I think this is unlikely to change given the purpose of `AbstractConst`.

Reply:
we don't want this, should never need an `AbstractConst` for mir constants. These are only used to unify constants in the type system